languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml

```diff
@@ -28745,4 +28607,609 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+		        <example correction="Um sich umzuziehen," type="incorrect"><marker>Um sich umzuziehen</marker> ging sie nach Hause.</example>
+		        <example><marker>Um sich umzuziehen,</marker> ging sie nach Hause.</example>
+		    </rule>
+	    </rulegroup>
+        <rulegroup id="KOMMAFEHLER" name="Fehler in der Kommasetzung">   
+		    <url>http://www.duden.de/sprachwissen/rechtschreibregeln/komma</url>
+		    <rule id="KOMMA_VOR_SONDERN" name="Komma vor 'sondern'">
+		        <pattern>
+					<marker>
+		            <token><exception regexp="yes">,|;|-|–|\(|„|“</exception></token>
+					<token>sondern</token>
+					</marker>
+		        </pattern>
+		        <message>Bitte ein Komma vor '\2' setzen!</message>
+	            <suggestion>\1, \2</suggestion>
+		        <example correction="Sommer, sondern" type="incorrect">Es ist nicht <marker>Sommer sondern</marker> Winter.</example>
+		        <example>Es ist nicht <marker>Sommer, sondern</marker> Winter.</example>
+		    </rule>
+		    <rule id="KOMMA_VOR_ERLAEUTERUNG" name="Komma vor Erläuterung" default="off">
+<!-- Sonderfälle noch nicht behandelt, deshalb deafult="off"  -->
+		        <pattern>
+					<marker>
+		            <token><exception regexp="yes">,|;|-|–|\(|„|“</exception></token>
+					<token regexp="yes">nämlich|insbesondere</token>
+					</marker>
+		        </pattern>
+		        <message>Bitte ein Komma vor '\2' setzen!</message>
+	            <suggestion>\1, \2</suggestion>
+		        <example correction="Jahreszeiten, nämlich" type="incorrect">Es gibt vier <marker>Jahreszeiten nämlich</marker> Frühling, Sommer, Herbst und Winter.</example>
+		        <example>Es gibt vier <marker>Jahreszeiten, nämlich</marker> Frühling, Sommer, Herbst und Winter.</example>
+		    </rule>
+		    <rule id="KOMMA_VOR_UND_ODER" name="Fehlerhaftes Komma vor 'und' oder 'oder'">
+		        <antipattern>
+					<marker>
+					<token skip="-1">,</token>
+					<token>,</token>
+					<token regexp="yes">und|oder</token>
+					</marker>
+		        </antipattern>
+		        <antipattern>
+					<marker>
+					<token>,</token>
+					<token>und</token>
+					<token>zwar</token>
+					</marker>
+		        </antipattern>
+		        <pattern>
+					<marker>
+					<token>,</token>
+					<token regexp="yes">und|oder</token>
+					</marker>
+		        </pattern>
+		        <message>Bitte das Komma vor und/oder löschen!</message>
+	            <suggestion> \2</suggestion>
+		        <example correction=" und" type="incorrect">Die Sonne versank hinter dem Horizont<marker>, und</marker> die Schatten der Nacht senkten sich über das Land.</example>
+		        <example>Die Sonne versank hinter dem Horizont<marker> und</marker> die Schatten der Nacht senkten sich über das Land.</example>
+		    </rule>
+		    <rule id="KOMMA_NACH_DIREKTER_REDE" name="Fehlendes Komma nach direkter Rede">
+		        <antipattern>
+					<token postag="SENT_START"/>
+					<token regexp="yes">»|«|"</token>
```

Comment:
> Ich frage mich jetzt: Ist es gesichert, dass die Sätze von der Kommasetzung einwandfrei sind? 

Nein, es ist im Gegenteil so gut wie sicher, dass viele Fehler drin sind. Gerade deswegen ist Tatoeba gut geeignet zum Testen. Man muss nur kritisch hinsehen und ggf. nachschlagen.

>Die Regel mit den Leerzeichen statt Tabulatoren kannte ich nicht. Sie ist mir aber sofort einleuchtend, da wahrscheinlich alle Entwickler unterschiedliche Editoren verwenden und es damit zu unterschiedlichen Layouts führt.

Danke für Dein Verständnis. Es ist übrigens keine in Stein gemeißelte Regel, sondern ein pragmatischer Konsens, den wir im deutschen Team mal gefunden haben. Wenn man durchgängig Tabs verwenden könnte, wäre es eigentlich eleganter, aber das wird nicht von allen Editoren unterstützt.